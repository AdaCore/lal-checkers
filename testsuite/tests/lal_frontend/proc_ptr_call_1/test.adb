procedure Test is
   X : Integer;

   procedure F is
   begin
      X := 12;
   end F;

   P : access procedure := F'Access;
begin
   X := 21;
end Test;
