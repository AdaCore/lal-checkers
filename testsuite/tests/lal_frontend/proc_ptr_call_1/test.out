Program:
  $stack#1 = Updated_1($stack#1, Input())
  P#2 = Subp_<SubpBody ["F"] test.adb:4:4-7:10>(Var_1($stack#1))
  assume(!=(P#2, null))
  ret0#3 = Call(P#2, Offset_3($stack#1))
  $stack#1 = CopyOffset($stack#1, Get_0(ret0#3))
  end:
